### ISSUE TITLE
Images with custom lima-init do not support provisioning scripts

### ISSUE THREAD

**Issue by simjnd** [lima-vm/alpine-lima#17]
When using the [`alpine-lima-std`](https://github.com/lima-vm/alpine-lima/releases/tag/v0.1.4) images, provisioning scripts don't seem to execute whereas they run fine on the old [`alpine-lima-ci`](https://github.com/lima-vm/alpine-lima/releases/tag/v0.0.1) images.

Try adding these provisioning scripts to Lima VMs running both the `std` and the `ci` versions:
```yaml
provision:
  - mode: system
    script: &setup-cache-and-repos |-
      #!/bin/sh
      set -eux -o pipefail
      setup-apkcache /Users/simjnd/.lima/_mounts/apk-cache
      cat <<EOF > /etc/apk/repositories
      http://alpine.42.fr/v3.13/main
      http://alpine.42.fr/v3.13/community
      http://alpine.42.fr/edge/main
      http://alpine.42.fr/edge/community
      http://alpine.42.fr/edge/testing
      EOF

  - mode: system
    script: &install-dependencies |-
      #!/bin/sh
      set -eux -o pipefail
      apk update
      apk add podman
```

After starting the `std` version, trying to run `podman` returns `sudo: podman: command not found` whereas the `ci` version returns `Error: missing command 'podman COMMAND'. Try 'podman --help' for more information.` indicating that podman has indeed been installed.

I assume in the title that this is due to the custom `lima-init` that is used instead of `cloud-init`?

**Comment by simjnd** [comment]
I am indeed using Lima 0.6.4, I had not played with Lima in a while and so I updated it before touching it again today
![image](https://user-images.githubusercontent.com/43272781/133837912-a497774a-fe70-4ae8-9e73-f64f473f8527.png)

Should I also re-open the issue on the lima repo?

**Comment by jandubois** [comment]
Sorry, I misread your description; I thought this was about user provisioning scripts, which were broken until last week, but this is actually about system scripts.

I see that the script fails during init processing (from `/var/log/lima-init.log`):

```
LIMA| Executing /mnt/lima-cidata/provision.system/00000000
+ setup-apkcache /Users/jan/.lima/_mounts/apk-cache
+ cat
LIMA| Executing /mnt/lima-cidata/provision.system/00000001
+ apk update
fetch http://alpine.42.fr/v3.13/main/x86_64/APKINDEX.tar.gz
ERROR: http://alpine.42.fr/v3.13/main: temporary error (try again later)
WARNING: Ignoring http://alpine.42.fr/v3.13/main: No such file or directory
fetch http://alpine.42.fr/v3.13/community/x86_64/APKINDEX.tar.gz
ERROR: http://alpine.42.fr/v3.13/community: temporary error (try again later)
WARNING: Ignoring http://alpine.42.fr/v3.13/community: No such file or directory
fetch http://alpine.42.fr/edge/main/x86_64/APKINDEX.tar.gz
ERROR: http://alpine.42.fr/edge/main: temporary error (try again later)
WARNING: Ignoring http://alpine.42.fr/edge/main: No such file or directory
fetch http://alpine.42.fr/edge/community/x86_64/APKINDEX.tar.gz
ERROR: http://alpine.42.fr/edge/community: temporary error (try again later)
WARNING: Ignoring http://alpine.42.fr/edge/community: No such file or directory
fetch http://alpine.42.fr/edge/testing/x86_64/APKINDEX.tar.gz
ERROR: http://alpine.42.fr/edge/testing: temporary error (try again later)
WARNING: Ignoring http://alpine.42.fr/edge/testing: No such file or directory
5 errors; 65 distinct packages available
LIMA| WARNING: Failed to execute /mnt/lima-cidata/provision.system/00000001
LIMA| Exiting with code 1
```

But I can run `apk update` successfully once the VM is fully booted:

```
lima-alpine:/Users/jan$ sudo apk update
fetch http://alpine.42.fr/v3.13/main/x86_64/APKINDEX.tar.gz
fetch http://alpine.42.fr/v3.13/community/x86_64/APKINDEX.tar.gz
fetch http://alpine.42.fr/edge/main/x86_64/APKINDEX.tar.gz
fetch http://alpine.42.fr/edge/community/x86_64/APKINDEX.tar.gz
fetch http://alpine.42.fr/edge/testing/x86_64/APKINDEX.tar.gz
v3.13.6-19-g03009d53f7 [http://alpine.42.fr/v3.13/main]
v3.13.6-21-g62a6abd4b7 [http://alpine.42.fr/v3.13/community]
v3.15.0_alpha20210804-2485-g1688b73ff4 [http://alpine.42.fr/edge/main]
v3.15.0_alpha20210804-2498-gc098c72de8 [http://alpine.42.fr/edge/community]
v3.15.0_alpha20210804-2497-g7ab2ba98e1 [http://alpine.42.fr/edge/testing]
OK: 33313 distinct packages available
```

So it looks like maybe the network is not fully up by the time the provisioning scripts run.

I'm re-opening this issue. The issue in the lima repo is unrelated and should remain closed.

**Comment by jandubois** [comment]
Fixed by #19
